src/ra_type_traits.h

```diff
@@ -125,9 +132,28 @@ _CONSTANT_VAR is_lessthan_comparable_v{detail::is_lessthan_comparable<LessThanCo
 template<typename LessThanComparable>
 _CONSTANT_VAR is_nothrow_lessthan_comparable_v{detail::is_nothrow_lessthan_comparable<LessThanComparable>::value};
 
-template<typename Comparable> _CONSTANT_VAR is_comparable_v{detail::is_comparable<Comparable>::value};
+template<typename Comparable>
+_CONSTANT_VAR is_comparable_v{detail::is_comparable<Comparable>::value};
+
+template<typename Comparable>
+_CONSTANT_VAR is_nothrow_comparable_v{detail::is_nothrow_comparable<Comparable>::value};
+
+namespace detail {
+
+template<typename T>
+struct _NODISCARD is_literal_type
+    : std::bool_constant<(std::is_scalar_v<T> || std::is_reference_v<T>) ||
+                         (std::is_aggregate_v<T> && std::is_scalar_v<typename T::value_type>) ||
+                         (std::is_class_v<T> && std::is_trivially_destructible_v<T> &&
+                          std::is_nothrow_constructible_v<T>)>
```

Comment:
According the standard a LiteralType is (Summarized):
- scalar type or
- reference type or
- an array of literal type or (array is an aggregate type)
- a class of literal type

I couldn't find a way to test for a constexpr constructor, union volatility, or closure type (lambda). 

The last condition should mostly be enough for our purposes. If the ones for `is_class_v` pass, it has a very high chance of having a constexpr constructor. We don't deal with device drivers or other type of hardware code so volatile shouldn't matter. Lambdas are _usually_ (not always) reference types, there is a `std::is_function_v`, but not all function types are literal types.